File: redsheet-portal/src/app/database-models/v-role-permission.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vRolePermission {
    constructor(vRolePermission?: vRolePermission )  {
        let properties = [ "active", "applicationModuleId", "canAdd", "canDelete", "canEdit", "canView", "expr1", "isRolePermissionItem", "isRoot", "moduleMasterActive", "moduleMasterName", "parentApplicationModuleId", "roleId", "roleName", "rolePermissionId", "visibleActionItem",];
        for (let property of properties)
            if (vRolePermission && vRolePermission[property])
                this[property] = vRolePermission[property];
    }
 
	active : number =   undefined;
 
	applicationModuleId : number =   0 ;
 
	canAdd : boolean = false ;
 
	canDelete : boolean = false ;
 
	canEdit : boolean = false ;
 
	canView : boolean = false ;
 
    @range(1,2147483647)
	expr1 : number =   undefined;
 
	isRolePermissionItem : boolean = false ;
 
	isRoot : boolean = false ;
 
	moduleMasterActive : boolean = false ;
 
    @required()
    @maxLength(100)
	moduleMasterName : string =   undefined;
 
	parentApplicationModuleId : number =   undefined;
 
	roleId : number =   undefined;
 
    @maxLength(50)
	roleName : string =   undefined;
 
	rolePermissionId : number =   undefined;
 
    @required()
    @maxLength(1)
	visibleActionItem : string =   undefined;


}

File: RedSheet.DbEntities/Models/vRolePermission.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vRolePermissions", Schema = "dbo")]
	[Serializable]
	public partial class vRolePermission  
	{

	public Nullable<int> Active { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int ApplicationModuleId { get; set; } 

	public Nullable<bool> CanAdd { get; set; } 

	public Nullable<bool> CanDelete { get; set; } 

	public Nullable<bool> CanEdit { get; set; } 

	public Nullable<bool> CanView { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int Expr1 { get; set; } 

	public Nullable<bool> IsRolePermissionItem { get; set; } 
 
	[Required]
	public bool IsRoot { get; set; } 
 
	[Required]
	public bool ModuleMasterActive { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string ModuleMasterName { get; set; } 

	public Nullable<int> ParentApplicationModuleId { get; set; } 

	public Nullable<int> RoleId { get; set; } 
 
	[MaxLength(50)]
	public string RoleName { get; set; } 

	public Nullable<int> RolePermissionId { get; set; } 
 
	[Required] 
	[MaxLength(1)]
	public string VisibleActionItem { get; set; } 

	public vRolePermission()
	{
    }

    }
}

